pkg/monitor/datapath_trace.go

```diff
@@ -77,9 +94,39 @@ func connState(reason uint8) string {
 	return fmt.Sprintf("%d", reason)
 }
 
+// DecodeTraceNotify will decode 'data' into the provided TraceNotify structure
+func DecodeTraceNotify(data []byte, tn *TraceNotify) error {
+	if len(data) < traceNotifyCommonLen {
+		return fmt.Errorf("Unknown trace event")
+	}
+
+	var msgHdrLen uint16
+	// metaLenOffset := unsafe.Offsetof(TraceNotify.MetaLen)
+	metaLenOffset := 14
+	// metaLenSize := unsafe.Sizeof(TraceNotify.MetaLen)
+	metaLenSize := 2
+	metaLen := data[metaLenOffset : metaLenOffset+metaLenSize]
+	err := binary.Read(bytes.NewReader(metaLen), byteorder.Native, &msgHdrLen)
+	if err != nil {
+		return err
+	}
+
+	switch msgHdrLen {
+	// Cilium v1.6 or earlier always reports 0 for the length.
+	case 0, traceNotifyV1Len:
+		err = binary.Read(bytes.NewReader(data), byteorder.Native, &tn.TraceNotifyV1)
+		tn.MetaLen = traceNotifyV1Len
+	case traceNotifyV2Len:
+		err = binary.Read(bytes.NewReader(data), byteorder.Native, tn)
+	default:
+		err = fmt.Errorf("Unrecognized trace event (len %d)", msgHdrLen)
+	}
+	return err
+}
+
 func (n *TraceNotify) encryptReason() string {
 	if (n.Reason & TraceReasonEncryptMask) != 0 {
-		return fmt.Sprintf("encrypted ")
+		return fmt.Sprintf(" encrypted ")
```

Comment:
Should the space be added somewhere else?

Reply:
Oh, I had another minor cosmetic fix that was supposed to go with this, but I dropped the other part from this PR and accidentally left this in.